api/cluster-config/v1/clusterconfig_types.go

```diff
@@ -25,26 +25,61 @@ import (
 
 // ClusterConfigSpec defines the desired state of ClusterConfig
 type ClusterConfigSpec struct {
+	//AdvertisementConfig defines the configuration for the advertisement protocol
 	AdvertisementConfig AdvertisementConfig `json:"advertisementConfig"`
 	DiscoveryConfig     DiscoveryConfig     `json:"discoveryConfig"`
 	LiqonetConfig       LiqonetConfig       `json:"liqonetConfig"`
 	DispatcherConfig    DispatcherConfig    `json:"dispatcherConfig,omitempty"`
 }
 
+//AdvertisementConfig defines the configuration for the advertisement protocol
 type AdvertisementConfig struct {
-	// +kubebuilder:validation:Maximum=100
-	// +kubebuilder:validation:Minimum=0
-	ResourceSharingPercentage int32 `json:"resourceSharingPercentage,omitempty"`
-	EnableBroadcaster         bool  `json:"enableBroadcaster,omitempty"`
-	// +kubebuilder:validation:Minimum=0
-	MaxAcceptableAdvertisement int32 `json:"maxAcceptableAdvertisement,omitempty"`
-	AutoAccept                 bool  `json:"autoAccept"`
+	//OutgoingConfig defines the behaviour for the creation of Advertisements on other clusters
+	OutgoingConfig BroadcasterConfig `json:"outgoingConfig"`
+	//IngoingConfig defines the behaviour for the acceptance of Advertisements from other clusters
+	IngoingConfig AdvOperatorConfig `json:"ingoingConfig"`
+	//KeepaliveThreshold defines the number of failed attempts to contact the foreign cluster your cluster will tolerate before deleting it.
 	// +kubebuilder:validation:Minimum=0
 	KeepaliveThreshold int32 `json:"keepaliveThreshold,omitempty"`
+	//After establishing a sharing with a foreign cluster, a keepalive mechanism starts, in order to know if the foreign cluster is reachable or not.
+	//KeepaliveRetryTime defines the time between an attempt to contact the foreign cluster and the next one.
 	// +kubebuilder:validation:Minimum=0
 	KeepaliveRetryTime int32 `json:"keepaliveRetryTime,omitempty"`
 }
 
+type BroadcasterConfig struct {
+	//ResourceSharingPercentage defines the percentage of your cluster resources that you will share with foreign clusters
+	// +kubebuilder:validation:Maximum=100
+	// +kubebuilder:validation:Minimum=0
+	ResourceSharingPercentage int32 `json:"resourceSharingPercentage"`
+	//EnableBroadcaster flag allows you to enable/disable the broadcasting of your Advertisement to the foreign clusters
+	EnableBroadcaster bool `json:"enableBroadcaster"`
+}
+
+// AcceptPolicy defines the policy to accept/refuse an Advertisement
+type AcceptPolicy string
+
+const (
+	// AutoAcceptWithinMaximum means all the Advertisement received will be accepted until the MaxAcceptableAdvertisement limit is reached
+	// AutoAcceptAll can be achieved by setting MaxAcceptableAdvertisement to infinite
```

Comment:
@fraborg Sorry for coming so late but... what's the value for "infinite" here?

Reply:
the maximum value of int32 is 2147483647, but we can set the MaxAcceptableAdvertisement range from 0 to a value we decide, which can be our "infinite"

Reply:
I would write this in the code (or, better, the definition of the structure), so that users know what they can configure.
Btw, not sure if we're telling users if this number refers to the max number of *contemporary* advertisements accepted, or the total number over time. Just in case, make sure this is clearly explained.